### DISCUSSION TITLE
pg_dumpall on the PostgreSQL database

### DISCUSSION THREAD

**Discussion by seethro** [nextcloud/all-in-one#818] [Questions] [answered]
Hello,

I love the Borg backup feature, but I'd really like to do a pg_dumpall on the db.

From my docker host, when I run:
`sudo docker exec -t nextcloud-aio-database pg_dumpall -c -U postgres > dump_aio_db.sql`

I get something like this in the .sql file:
`pg_dumpall: error: connection to server on socket "/var/run/postgresql/.s.PGSQL.5432" failed: FATAL:  role "postgres" does not exist
`

I'm a docker noob so please be gentle...

**Comment by szaimen** [comment, marked as answer]
Simply run this inside the database container?
https://github.com/nextcloud/all-in-one/blob/bad431984ec86468a1d6cc8eceb8f88851c131a2/Containers/postgresql/start.sh#L114

**Reply by seethro** [reply]
Thanks for the tip... looks like I was using the wrong user!!
This worked for me:
`docker exec -it nextcloud-aio-database pg_dump -U nextcloud nextcloud_database`
